base/abstractset.jl

```diff
@@ -375,6 +377,27 @@ function issetequal(l, r)
     return issetequal(Set(l), Set(r))
 end
 
+## set disjoint comparison
+"""
+    isdisjoint(v1, v2) -> Bool
+
+Returns whether the collections `v1` and `v2` are disjoint, i.e. whether
+their intersection is empty.
+"""
+function isdisjoint(l, r)
+    function _isdisjoint(l, r)
+        hasfastin(r) && return !any(in(r), l)
+        hasfastin(l) && return !any(in(l), r)
+        haslength(r) && length(r) < FASTIN_SET_THRESHOLD &&
+            return !any(in(r), l)
+        return !any(in(Set(r)), l)
+    end
+    if haslength(l) && haslength(r) && lenght(r) < length(l)
```

Comment:
Sometimes `lenght` is not constant time (e.g. `BitSet`), and in some cases it might be computed twice here. Not a big deal at all (I don't have any example where this is an actual problem (this requires `haslengh`, `!hasfastin` and non-fast `length`)), but if this can be easily re-arranged to not compute the length twice, why not!

Reply:
I think we generally assume that `length` is constant time. Otherwise these checks are unlikely to be profitable. I don't think it's worth the extra complication. We're already doing a quadratic operation here.